services/wireguard/resources/allocator.go

```diff
@@ -17,29 +17,131 @@
 
 package resources
 
-import "net"
+import (
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+const prefix = "myst"
 
 // Allocator is mock wireguard resource handler.
 // It will manage lists of network interfaces names, IP addresses and port for endpoints.
-type Allocator struct{}
+type Allocator struct {
+	Ifaces      map[int]struct{}
+	IPAddresses map[int]struct{}
+	Port        map[int]struct{}
+	mu          sync.Mutex
+}
+
+// NewAllocator creates new resource pool for wireguard connection.
+func NewAllocator() Allocator {
+	return Allocator{
+		Ifaces:      make(map[int]struct{}),
+		IPAddresses: make(map[int]struct{}),
+		Port:        make(map[int]struct{}),
+	}
+}
 
 // AllocateInterface provides available name for the wireguard network interface.
-func (a *Allocator) AllocateInterface() string { return "myst0" }
+func (a *Allocator) AllocateInterface() string {
```

Comment:
I think it should return an error. Basically "" interface name is an error. Allocation may fail triggering critical error.

Reply:
Added error.